hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -501,7 +501,8 @@ public DeadServer getDeadServers() {
    * @return true if any RS are being processed as dead, false if not
    */
   public boolean areDeadServersInProgress() throws IOException {
-    return master.getProcedures().stream().anyMatch(p -> p instanceof ServerCrashProcedure);
+    return master.getMasterProcedureExecutor().getActiveProceduresNoCopy().stream()
```

Comment:
The getActiveProceduresNoCopy is only expected to be called when initialization, where we can make sure there is no race.